itests/api/service_accounts_test.py

```diff
@@ -1,69 +1,74 @@
 
 
-def test_get_service_accounts(api_client, users, service_accounts):  # noqa: F811
-    role_users = [username for username, u in users.items() if u.role_user]
-    assert len(role_users) > 0
+def test_get_service_accounts(tmpdir: LocalPath, setup: SetupTest) -> None:
+    with setup.transaction():
+        setup.create_role_user("role@a.co")
+        setup.create_service_account("service@a.co", "team-sre")
 
-    expected = role_users + list(service_accounts.keys())
+    with api_server(tmpdir) as api_url:
+        api_client = Groupy(api_url)
+        assert sorted(api_client.service_accounts) == ["role@a.co", "service@a.co"]
 
-    api_service_accounts = list(api_client.service_accounts)
-    assert sorted(api_service_accounts) == sorted(expected)
 
+def test_get_service_account(tmpdir: LocalPath, setup: SetupTest) -> None:
+    with setup.transaction():
+        setup.create_service_account(
+            "service@a.co",
+            owner="team-sre",
+            machine_set="some machines",
+            description="some service account",
+        )
 
-def test_get_service_account(api_client):  # noqa: F811
-    service_account = api_client.service_accounts.get("service@a.co")
-    assert service_account.groups == {}
-    assert service_account.passwords == []
-    assert service_account.public_keys == []
-    assert service_account.enabled
-    assert service_account.service_account == {
-        "description": "some service account",
-        "machine_set": "some machines",
-        "owner": "team-sre",
-    }
-    assert service_account.permissions == []
-    assert service_account.metadata == {}
+    with api_server(tmpdir) as api_url:
+        api_client = Groupy(api_url)
+        service_account = api_client.service_accounts.get("service@a.co")
+        assert service_account.groups == {}
+        assert service_account.passwords == []
+        assert service_account.public_keys == []
+        assert service_account.enabled
+        assert service_account.service_account == {
+            "description": "some service account",
+            "machine_set": "some machines",
+            "owner": "team-sre",
+        }
+        assert service_account.permissions == []
+        assert service_account.metadata == {}
 
 
-def test_get_role_user(api_client):  # noqa: F811
-    role_user = api_client.service_accounts.get("role@a.co")
-    assert role_user.groups == {}
-    assert role_user.passwords == []
-    assert role_user.public_keys == []
-    assert role_user.enabled
-    assert role_user.service_account is None
-    assert role_user.permissions == []
-    assert role_user.metadata == {}
+def test_get_role_user(tmpdir: LocalPath, setup: SetupTest) -> None:
+    with setup.transaction():
+        setup.create_role_user("role@a.co")
+
+    with api_server(tmpdir) as api_url:
+        api_client = Groupy(api_url)
+        role_user = api_client.service_accounts.get("role@a.co")
+        assert sorted(role_user.groups) == ["role@a.co"]
```

Comment:
Huh, why's this different? Was it just a bug in the old fixtures? That `tests/fixtures.py` code does seem suspect, though I haven't dug in -- if it is messed up could we also fix it since it isn't removed here?

Reply:
It's a bug in `tests/fixtures.py` that didn't create role users properly. I fixed that bug, but the fix is in some later PR that was stacked behind this one.